src/LuaTransformer.ts

```diff
@@ -5113,4 +5118,37 @@ export class LuaTransformer {
             return visitResult;
         }
     }
+
+    public generateConstructorDecorationStatement(
```

Comment:
I think `generate` is kind of out of place there, usually such methods start with `transform` or `create`.